operators/postgresql-operator/PostgresqlService.js

```diff
@@ -0,0 +1,187 @@
+      .then(ips => this.agent.updateDb(ips, this.guid, this.parameters))
+      .tap(() => logger.info(`+-> Updated logical db '${this.guid}' for deployment '${this.deploymentName}'`));
+  }
+
+  delete(changedOptions) {
+    const metadata = changedOptions.operatorMetadata;
+    const operation = {
+      type: 'delete',
+      parameters: {
+        deploymentName: metadata.dedicatedInstanceDeploymentName
+      }
+    };
+    let instanceDeleted = true;
+    return this.initialize(operation)
+      .tap(() => logger.info(`Deleting logical db '${this.guid}' for deployment '${this.deploymentName}'...`))
+      .then(() => this.director.getDeploymentIps(this.deploymentName))
+      .tap(() => instanceDeleted = false)
+      .then(ips => Promise.all([this.agent.deleteDb(ips, this.guid)]))
+      .tap(() => logger.info(`+-> Deleted logical db '${this.guid}' for deployment '${this.deploymentName}'`))
+      .catch(NotFound, () => {
+        throw new Gone(this.guid);
+      });
+  }
+
+  bind(params) {
+    const operation = {
+      type: 'bind'
+    };
+    return this.initialize(operation)
+      .then(() => this.createBinding(this.deploymentName, this.guid, {
+        id: params.binding_id,
+        parameters: params.parameters || {}
+      }));
+  }
+
+  createBinding(deploymentName, instanceId, binding) {
+    logger.info(`Creating binding '${binding.id}' with binding parameters '${binding.parameters}' for deployment '${deploymentName}', logicaldb instance '${instanceId}'...`);
+    return this.director.getDeploymentIps(deploymentName)
+      .then(ips => this.agent.createCredentials(ips, instanceId, binding.parameters))
+      .then(credentials => this.createBindingProperty(deploymentName, binding.id, _.set(binding, 'credentials', credentials)))
+      .then(() => binding.credentials)
+      .tap(() => {
+        const bindCreds = _.cloneDeep(binding.credentials);
+        utils.maskSensitiveInfo(bindCreds);
+        logger.info(`+-> Created binding:${JSON.stringify(bindCreds)}`);
+      });
+  }
+
+  unbind(params) {
+    const operation = {
+      type: 'unbind'
+    };
+    return this.initialize(operation)
+      .then(() => this.deleteBinding(this.deploymentName, this.guid, params.binding_id));
+  }
+
+  deleteBinding(deploymentName, instanceId, bindingId) {
+    logger.info(`Deleting binding '${bindingId}' for deployment '${deploymentName}' , logicaldb instance '${instanceId}'...`);
+    return Promise
+      .all([
+        this.director.getDeploymentIps(deploymentName),
+        this.getBindingProperty(deploymentName, bindingId)
+      ])
+      .spread((ips, binding) => this.agent.deleteCredentials(ips, instanceId, binding.credentials))
+      .then(() => this.deleteBindingProperty(deploymentName, bindingId))
+      .tap(() => logger.info('+-> Deleted service binding'))
+      .catchThrow(NotFound, new ServiceBindingNotFound(bindingId));
+  }
+
+  createBindingProperty(deploymentName, bindingId, value) {
```

Comment:
Why do you need it? We do not store binding property anymore in the bosh deployment and don't recomend it as well.